packages/legend-application-studio/src/stores/editor-state/element-editor-state/mapping/MappingElementDecorator.ts

```diff
@@ -323,10 +311,15 @@ export class MappingElementDecorator implements SetImplementationVisitor<void> {
     pureInstanceSetImpl_setPropertyMappings(
       setImplementation,
       decoratedPropertyMappings.concat(
-        propertyMappingsBeforeDecoration.filter(
-          (propertyMapping) =>
-            !decoratedPropertyMappings.includes(propertyMapping),
-        ),
+        propertyMappingsBeforeDecoration
+          .filter(
```

Comment:
I would like us to document this a bit, this is a fair optimization, but in the same spirit as this PR, we might in the future would like to pertain these properties instead